uvicorn/server.py

```diff
@@ -153,6 +155,9 @@ def _share_socket(sock: socket) -> socket:
                 logger.error(exc)
                 await self.lifespan.shutdown()
                 sys.exit(1)
+
+            assert server.sockets is not None
```

Comment:
is it a real possibility ?

Reply:
Mostly to please mypy here, as indeed `server.sockets` is an `Optional[list]`.